src/main/java/frc/team997/lib/trajectory/MotionProfile.java

```diff
@@ -0,0 +1,176 @@
+ * out of constraints and desired states.
+ */
+public class MotionProfile {
+    protected List<ProfilePhase> phases;
+    private State initialState;
+
+    /** Holds position and velocity outputs from a MotionProfile. */
+    public static class State {
+        public final double position;
+        public final double velocity;
+
+        /**
+         * Constructs a State out of a given position and velocity.
+         *
+         * @param position Position of the profile.
+         * @param velocity Velocity of the profile.
+         */
+        public State(double position, double velocity) {
+            this.position = position;
+            this.velocity = velocity;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            double epsilon = 0.0001;
+            if (other instanceof State) {
+                State rhs = (State) other;
+                return Math.abs(this.position - rhs.position) < epsilon
+                        && Math.abs(this.velocity - rhs.velocity) < epsilon;
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(position, velocity);
+        }
+
+        @Override
+        public String toString() {
+            return "State[position: " + position + ", velocity: " + velocity + ", time: ";
+        }
+    }
+
+    /**
+     * Constructor for a MotionProfile.
+     *
+     * @param initialState The initial state of the motion profile.
+     * @param phases The ProfilePhases to start the profile with.
+     */
+    public MotionProfile(State initialState, ProfilePhase... phases) {
+        this.initialState = initialState;
+        this.phases = new ArrayList<>();
+        for (ProfilePhase phase : phases) {
+            this.phases.add(phase);
+        }
+    }
+
+    /**
+     * Constructor for a MotionProfile with no phases and a defined initial state.
+     *
+     * <p>This constructor is protected because it encourages users to create empty profiles and
+     * then add phases to the (also protected) {@code phases}, when this can cause undefined
+     * behavior. This needs to exist, however, for certain subclasses of this to work properly.
+     *
+     * @param initialState The initial state of the motion profile.
+     */
+    protected MotionProfile(State initialState) {
+        this.initialState = initialState;
+        this.phases = new ArrayList<>();
+    }
+
+    /**
+     * Constructor for a MotionProfile with an initial state of zero displacement and zero velocity.
+     *
+     * @param phases The ProfilePhases to start the profile with.
+     */
+    public MotionProfile(ProfilePhase... phases) {
+        this(new State(0, 0), phases);
+    }
+
+    /**
+     * Gets all the phases in the profile.
+     *
+     * @return An ordered list of ProfilePhases.
+     */
+    public List<ProfilePhase> getPhases() {
```

Comment:
Do we have an agreed style for where getters and setters should go? I like grouping getters and setter with the corresponding variable but some people also put all the getters and setters together

Reply:
Not really? It's just personal preference I guess-- I tend to put getters/setters of the same data type together